http_client_wrapper.go

```diff
@@ -71,15 +80,28 @@ func (h httpClientWrapper) ExecuteRequest(path, method string, body []byte) (*ht
 			if err != nil {
 				return errors.Wrapf(err, "error reading failed request body")
 			}
-			errResp := errors.Errorf("received non success status code %d with body: %s", res.StatusCode, string(body))
+			errResp := BuildResponseError(res.StatusCode, errors.Errorf("received non success status code %d with body: %s", res.StatusCode, string(body)))
 			if res.StatusCode == 400 || res.StatusCode == 401 || res.StatusCode == 404 {
 				// unrecoverable since probably bad payload or n
-				return retry.Unrecoverable(errResp)
+				return retry.Unrecoverable(BuildResponseError(res.StatusCode, errResp))
 			}
 			return errResp
 		}
 		return err
 	}, retry.Attempts(5), retry.Delay(500*time.Millisecond), retry.MaxDelay(9*time.Second))
 
+	if res == nil {
+		return nil, err
+	}
+
+	if _, ok := err.(retry.Error); ok {
+		retryErrors := err.(retry.Error)
+		for _, e := range retryErrors {
+			if httpError, ok := e.(HTTPResponseError); ok {
```

Comment:
**thought:** This library is interesting and I hadn't looked at it closely before—if it's not `nil` the error here is a list of length `5` with all the errors returned. That is cool. I wonder if this is the right thing to do when, for example, we get a `500` and then later in the retry process a transport-level error.

I guess it's hard to summarize five different errors. What's here is good.